internal/controller/config.go

```diff
@@ -59,3 +61,16 @@ func secretName(sa *corev1.ServiceAccount) string {
 
 	return name
 }
+
+func secretNameIndexed(sa *corev1.ServiceAccount, idx int) string {
+	if idx <= 0 {
+		return secretName(sa)
+	}
+	base := secretName(sa)
+	suffix := fmt.Sprintf("-%d", idx)
+	max := validation.DNS1123SubdomainMaxLength
+	if len(base)+len(suffix) > max {
+		base = base[:max-len(suffix)]
```

Comment:
IMO introducing a new variable has little merit here.

```suggestion
	if len(base)+len(suffix) > validation.DNS1123SubdomainMaxLength {
		base = base[:validation.DNS1123SubdomainMaxLength-len(suffix)]
```